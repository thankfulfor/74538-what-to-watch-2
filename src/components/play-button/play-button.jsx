import * as React from 'react';
import PropTypes from 'prop-types';

export class PlayButton extends React.PureComponent {

  render() {
    const {onShowPlayerButtonClick} = this.props;

    return (
      <button onClick={onShowPlayerButtonClick} className="btn btn--play movie-card__button" type="button">
        <svg viewBox="0 0 19 19" width="19" height="19">
          <use xlinkHref="#play" />
        </svg>
        <span>Play</span>
      </button>
    );
  }
}

PlayButton.propTypes = {
  onShowPlayerButtonClick: PropTypes.func.isRequired,
};

