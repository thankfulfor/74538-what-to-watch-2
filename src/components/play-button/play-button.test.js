import React from 'react';
import renderer from 'react-test-renderer';
import {PlayButton} from './play-button';

it(`PlayButton корректно рендерится после перезапуска`, () => {
  const clickHandler = jest.fn();
  const tree = renderer
    .create(<PlayButton onShowPlayerButtonClick={clickHandler} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
