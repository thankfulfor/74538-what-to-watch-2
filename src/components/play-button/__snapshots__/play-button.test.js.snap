// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton корректно рендерится после перезапуска 1`] = `
<button
  className="btn btn--play movie-card__button"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    height="19"
    viewBox="0 0 19 19"
    width="19"
  >
    <use
      xlinkHref="#play"
    />
  </svg>
  <span>
    Play
  </span>
</button>
`;
